### PR TITLE
Entrepreneur trial plan page

### PR DESCRIPTION
**Pull request by andregardi**
## Proposed Changes

image
We are introducing the new Entrepreneur trial plan, which share some similarities with Woo Express trial, but requires customized experience on Plans page.

Unlike the Woo Express trial, we are not going to offer the 1U$ offer when upgrading the Entrepreneur trial.

## Testing Instructions

For testing this you will need an Entrepreneur trial account:
p1712672819187649-slack-C02TCEHP3HA

You will also need a Woo Express trial to check that we didn't break the original experience
https://woocommerce.com/start/

Access `http://calypso.localhost:3000/plans/{your-site-here}`
- Check that the trial banner does not refer to the 1U$ offer.
- Click on **Add payment method**.
- You should be redirected to checkout page targeting the Entrepreneur plan and the 1U$ offer should **not** be offered.
- Go back to the `http://calypso.localhost:3000/plans/{your-site-here}`.
- On the bottom of the page there should be the **Keep all the features for your site** section.
- Select different billing periods, this should update the price of the plan inside the card.
- Click on the second button **Add payment method** button, the one inside the card.
- The checkout page billing period should match the one you selected before.

Do a regression test with Woo Express trial account and check that:
- The 1U$ offer is presented on the banner and checkout page.
- Checkout page should target `Woo Express: Performance` plan instead of `Entrepreneur.`

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by ivan-ottinger** [comment]
Just noting here that I am reviewing this PR.

File: client/my-sites/plans/ecommerce-trial/entrepreneur-plan/entrepreneur-plan.jsx
Comment (ivan-ottinger) [inline client/my-sites/plans/ecommerce-trial/entrepreneur-plan/entrepreneur-plan.jsx]: This is probably coming from the design, but just wanted to check: don't we rather use `payment plan` here? (the same as we currently have in the CTA)

```suggestion
								"Continue enjoying the full benefits of Entrepreneur plan, simply add your payment method and maximize your store's potential."
```

Markup on 2024-04-23 at 16:53:06

File: client/my-sites/plans/trials/trial-banner/use-banner-subtitle.ts
Comment (ivan-ottinger) [inline client/my-sites/plans/trials/trial-banner/use-banner-subtitle.ts:78]: I think we could consider adding a `//translators:` comment for translators explaining what the variable in the string is - as they may not see the variable names (ref: PCYsg-3hI-p2).

This applies to other `translate()` calls with string variables.

File: client/my-sites/plans/ecommerce-trial/index.tsx
Comment (ivan-ottinger) [inline client/my-sites/plans/ecommerce-trial/index.tsx]: Couldn't we just pass `isEntrepreneurTrial` here - since we already calculated it above? But maybe I am missing something. 🤔 
```suggestion
								isEntrepreneurTrial={ isEntrepreneurTrial }
```

File: client/my-sites/plans/ecommerce-trial/index.tsx
Comment (ivan-ottinger) [inline client/my-sites/plans/ecommerce-trial/index.tsx]:
```suggestion
						isEntrepreneurTrial ? (
```

File: client/my-sites/plans/ecommerce-trial/entrepreneur-plan/style.scss
Comment (ivan-ottinger) [inline client/my-sites/plans/ecommerce-trial/entrepreneur-plan/style.scss]: I wonder if we don't have predefined variable for this size.

File: client/my-sites/plans/ecommerce-trial/entrepreneur-plan/style.scss
Comment (ivan-ottinger) [inline client/my-sites/plans/ecommerce-trial/entrepreneur-plan/style.scss]: I wonder if we don't have predefined variable for this size.

**Review by ivan-ottinger — COMMENTED** [review]
Great work, André! The proposed changes work well. 👌🏼🙂 

I wasn't able to review the whole PR yet and will continue tomorrow.

Details:
- the trial banner doesn't display the USD 1 offer
- when clicking the top CTA "Add payment method" button, redirection to the Checkout page works well; the Monthly Entrepreneur plan is added to the cart and the USD 1 offer is not offered
- "Keep all the features for your site" section is displayed on the bottom of the Plan page.
- selecting a different billing period updates the plan inside the shopping cart when the bottom CTA "Add payment method" button is clicked; this dropdown doesn't have effect on the top CTA button; I believe this is expected behavior

So far I only noticed couple minor things (some of them mentioned inline in the code).

**Billing period details**

When selecting the Monthly billing period, I missed more information a little bit:

| Monthly | 3-Year plan |
|--------|--------|
| Markup on 2024-04-23 at 16:59:17 | Markup on 2024-04-23 at 16:59:45 | 

Maybe this is intentional - as most information isn't relevant there. Have we considered displaying at least "per month, excl. taxes"?

File: client/my-sites/plans/ecommerce-trial/entrepreneur-plan/entrepreneur-plan.jsx
Comment (mashikag) [reply client/my-sites/plans/ecommerce-trial/entrepreneur-plan/entrepreneur-plan.jsx]: Actually following up on this, we may consider changing the CTA copy and the description altogether. What we are asking the user is to actually purchase a full plan, rather than to enter a payment method only. 
Related: p1713898784084409/1713529322.545209-slack-C06RQ4EQ7V2

Comment (mashikag) [reply client/my-sites/plans/ecommerce-trial/entrepreneur-plan/entrepreneur-plan.jsx]: However, this should not be PR blocking. We can merge in before deciding on the copies. Perhaps we can just refrain from submitting those for translation.

**Comment by mashikag** [comment]
I see some spacing differences between the different sections in comparison to the designs. 
Screenshot 2024-04-23 at 21 14 32

File: client/my-sites/plans/ecommerce-trial/entrepreneur-plan/entrepreneur-plan.jsx
Comment (andregardi) [reply client/my-sites/plans/ecommerce-trial/entrepreneur-plan/entrepreneur-plan.jsx]: Thank you, I addressed it. And kept the button text out of the translate context.

File: client/my-sites/plans/trials/trial-banner/use-banner-subtitle.ts
Comment (andregardi) [reply client/my-sites/plans/trials/trial-banner/use-banner-subtitle.ts:78]: Sure sure!
I see we also have a pattern in which comment is a parameter of the translation function. I am using it.

File: client/my-sites/plans/ecommerce-trial/index.tsx
Comment (andregardi) [reply client/my-sites/plans/ecommerce-trial/index.tsx]: Thank you for pointing that. This condition was introduced before I created the isEntrepreneurTrial and I didn't remember to update that.

File: client/my-sites/plans/ecommerce-trial/index.tsx
Comment (andregardi) [reply client/my-sites/plans/ecommerce-trial/index.tsx]: Yeah! Same as above.

File: client/my-sites/plans/ecommerce-trial/entrepreneur-plan/style.scss
Comment (andregardi) [reply client/my-sites/plans/ecommerce-trial/entrepreneur-plan/style.scss]: Got it, it is called `font-title-medium`

File: client/my-sites/plans/ecommerce-trial/entrepreneur-plan/style.scss
Comment (andregardi) [reply client/my-sites/plans/ecommerce-trial/entrepreneur-plan/style.scss]: Got it, it is called `$font-body-small`

**Comment by andregardi** [comment]
> I see some spacing differences between the different sections in comparison to the designs. Screenshot 2024-04-23 at 21 14 32

Thank you for catching it. It is fixed now.

File: client/my-sites/plans/trials/trial-banner/use-banner-subtitle.ts
Comment (ivan-ottinger) [reply client/my-sites/plans/trials/trial-banner/use-banner-subtitle.ts:78]: If I am understanding correctly, we are planning to address this in another PR, right?

I am asking because I see you have added the comment to the `translate()` call below already, so maybe this one was left out accidentally. 🙂

File: client/my-sites/plans/current-plan/trials/use-go-to-checkout-with-plan.tsx
Comment (ivan-ottinger) [inline client/my-sites/plans/current-plan/trials/use-go-to-checkout-with-plan.tsx:40]:
```suggestion
	 * Redirects to the checkout page with Plan in cart.
```

File: client/my-sites/plans/current-plan/trials/use-go-to-checkout-with-plan.tsx
Comment (ivan-ottinger) [inline client/my-sites/plans/current-plan/trials/use-go-to-checkout-with-plan.tsx:30]: This question comes from the background I am missing and curiosity: Why are we getting these two values differently, i.e. why we cannot get them the same way? 🤔 

I understand that there's a chance the answer might not be straightforward so we can discuss it in chat or a call.

File: client/my-sites/plans/components/upgrade-button/style.scss
Comment (ivan-ottinger) [inline client/my-sites/plans/components/upgrade-button/style.scss:1]: I know that this is not uniformly followed through the whole Calypso repo, but maybe we could consider naming the class based on the component / filename (ref. https://github.com/Automattic/wp-calypso/blob/trunk/docs/coding-guidelines/css.md):

> This is an example of a declaration:
> 
> `.component__element.is-modifier {}`
> 
> Two important considerations:
> 
> - The component fragment matches the folder name of the React component it's providing styles to.
> [...]

File: client/my-sites/plans/components/upgrade-button/upgrade-button.tsx
Comment (ivan-ottinger) [inline client/my-sites/plans/components/upgrade-button/upgrade-button.tsx:14]: Just marking this in case we decide to remove the `translation` call before merging - until we decide for the right copy.

**Review by ivan-ottinger — APPROVED** [review]
Thank you for the updates, @andregardi. 🙂 

I did a second review / test round and the changes work as expected. 👍🏼👍🏼 I didn't notice any regressions in the Woo Express Plan(s) page either. 

There are a couple small additional things I noticed, but as we discussed in p1713910018466149/1713821383.190949-slack-C02TCEHP3HA, I am approving the PR so we can address them in subsequents PRs.

File: client/my-sites/plans/current-plan/trials/use-go-to-checkout-with-plan.tsx
Comment (andregardi) [reply client/my-sites/plans/current-plan/trials/use-go-to-checkout-with-plan.tsx:30]: Both Entrepreneur and Woo Express trial share the same product, `PLAN_ECOMMERCE_TRIAL_MONTHLY`, and on differ on billing plans.

File: client/my-sites/plans/trials/trial-banner/use-banner-subtitle.ts
Comment (emilyaudela) [inline client/my-sites/plans/trials/trial-banner/use-banner-subtitle.ts:87]: Would it be possible to add plural here? (1 day, 2 days, etc.) Some languages have multiple plurals, so this would be helpful in getting an accurate translation.

File: client/my-sites/plans/ecommerce-trial/entrepreneur-plan/entrepreneur-plan.jsx
Comment (ivan-ottinger) [reply client/my-sites/plans/ecommerce-trial/entrepreneur-plan/entrepreneur-plan.jsx]: The copy is being addressed in https://github.com/Automattic/wp-calypso/pull/89878.

File: client/my-sites/plans/current-plan/trials/use-go-to-checkout-with-plan.tsx
Comment (ivan-ottinger) [reply client/my-sites/plans/current-plan/trials/use-go-to-checkout-with-plan.tsx:40]: This is being addressed in https://github.com/Automattic/wp-calypso/pull/89878.

File: client/my-sites/plans/components/upgrade-button/style.scss
Comment (ivan-ottinger) [reply client/my-sites/plans/components/upgrade-button/style.scss:1]: This is being addressed in https://github.com/Automattic/wp-calypso/pull/89878.

File: client/my-sites/plans/components/upgrade-button/upgrade-button.tsx
Comment (ivan-ottinger) [reply client/my-sites/plans/components/upgrade-button/upgrade-button.tsx:14]: This is being addressed in https://github.com/Automattic/wp-calypso/pull/89878.

File: client/my-sites/plans/trials/trial-banner/use-banner-subtitle.ts
Comment (ivan-ottinger) [reply client/my-sites/plans/trials/trial-banner/use-banner-subtitle.ts:87]: Nice catch! This is being addressed in https://github.com/Automattic/wp-calypso/pull/89878.

**Pull request merged**
